src/core/vector/qgsvectorlayerfeaturecounter.h

```diff
@@ -95,8 +97,9 @@ class CORE_EXPORT QgsVectorLayerFeatureCounter : public QgsTask
     QgsExpressionContext mExpressionContext;
     QHash<QString, long> mSymbolFeatureCountMap;
     QHash<QString, QgsFeatureIds> mSymbolFeatureIdMap;
+    std::unique_ptr< QgsFeedback > mFeedback;
     bool mWithFids = false;
-    int mFeatureCount;
+    long mFeatureCount = 0;
```

Comment:
For Win64:
```suggestion
    qint64 mFeatureCount = 0;
```

Reply:
I've matched it to QgsVectorLayer::featureCount()/QgsVectorDataProvider::featureCount(), which all use longs.

Reply:
> I've matched it to QgsVectorLayer::featureCount()/QgsVectorDataProvider::featureCount(), which all use longs.

ah, they are wrong then. I'll send a PR to fix them

Reply:
Can I ask what the issue is with the long type? It's been this way for decades :laughing:

Reply:
> Can I ask what the issue is with the long type? It's been this way for decades laughing

That's a common portability issue: long is 32-bit on Windows 64 bit, whereas it is 64-bit on Linux 64 bit : https://stackoverflow.com/questions/384502/what-is-the-bit-size-of-long-on-64-bit-windows

Reply:
Could we just go "long long" instead then. and avoid the qt type?

Reply:
that can do